File: src/main/resources/static/js/biodata.js
var tableBiodata = {
    create: function () {
        // jika table tersebut datatable, maka clear and destroy
        if ($.fn.DataTable.isDataTable('#tableBiodata')) {
            //table yg sudah dibentuk menjadi datatable harus di-rebuild lagi untuk di-instantiasi-ulang
            $('#tableBiodata').DataTable().clear();
            $('#tableBiodata').DataTable().destroy();
        }

        $.ajax({
            url: '/api/biodata',
            method: 'get',
            contentType: 'application/json',
            success: function (res, status, xhr) {
                if (xhr.status == 200 || xhr.status == 201) {
                    $('#tableBiodata').DataTable({
                        data: res,
                        columns: [
                               {
                                   title: "Full Name",
                                   data: "fullName"
                               },
							   {
								   title: "Email",
								   data: "email"
							   },
							   {
								   title: "Phone Number",
								   data: "phone"
							   },
							   {
								   title: "Adress",
								   data: "address"
							   },
							   {
								   title: "Birthday",
								   data: "birthday"
							   },
							   {
								   title: "ID Card Number",
								   data: "idCard"
							   },
							   {
								   title: "University",
								   data: "univ"
							   },
							   {
								   title: "Major",
								   data: "major"
							   },
							   {
								   title: "GPA",
								   data: "gpa"
							   },
							   {
								   title: "Cover Letter",
								   data: "cover"
							   },
							   {
								   title: "Hobby",
								   data: "hobby"
							   },
							   {
								   title: "Gender",
								   data: "gender"
							   },
							   {
								   title: "Religion",
								   data: "religion"
							   },
                               {
                                title: "Action",
                                data: null,
                                render: function (data, type, row) {
                                    return "<button class='btn-primary' onclick=formBiodata.setEditData('" + data.id + "')>Edit</button>"
                                }
                            }
                        ]
                    });

                } else {

                }
            },
            erorrr: function (err) {
                console.log(err);
            }
        });


    }
};

var formBiodata = {
    resetForm: function () {
        $('#form-biodata')[0].reset();
    },
    saveForm: function () {
    	console.log('asdasd');
        if ($('#form-biodata').parsley().validate()) {
            var dataResult = getJsonForm($("#form-biodata").serializeArray(), true);
//            document.getElementById("demo").innerHTML = JSON.stringify(dataResult);

            $.ajax({
                url: '/api/biodata',
                method: 'post',
                contentType: 'application/json',
                dataType: 'json',
                data: JSON.stringify(dataResult),
                success: function (res, status, xhr) {
                    if (xhr.status == 200 || xhr.status == 201) {
                        tableBiodata.create();
                        $('#modal-biodata').modal('hide')

                    } else {

                    }
                },
                erorrr: function (err) {
                    console.log(err);
                }
            });
        }
    }, setEditData: function (idCabang) {
        formBiodata.resetForm();

        $.ajax({
            url: '/api/biodata' + idCabang,
            method: 'get',
            contentType: 'application/json',
            dataType: 'json',
            success: function (res, status, xhr) {
                if (xhr.status == 200 || xhr.status == 201) {
                    $('#form-biodata').fromJSON(JSON.stringify(res));
                    $('#modal-biodata').modal('show')

                } else {

                }
            },
            erorrr: function (err) {
                console.log(err);
            }
        });


    }

};

File: src/main/java/com/vinan/myfirstthymeleaf/model/entity/Biodata.java
package com.vinan.myfirstthymeleaf.model.entity;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

import javax.persistence.*;

@Entity
@Table(name = Biodata.TABLE_NAME)
@Data
@Getter @Setter
public class Biodata {
    public static final String TABLE_NAME = "t_biodata";

    @Id
    @GeneratedValue
    @Column(name = "id_bio")
    private Integer id;
    @Column(name = "name", nullable = false)
    private String fullName;
    @Column(name = "email")
	private String email;
    @Column(name = "phone_number")
	private String phone;
    @Column(name = "address")
	private String address;
    @Column(name = "birthday")
	private Date birthday;
    @Column(name = "id_card_number")
	private String idCard;
    @Column(name = "university")
	private String univ;
    @Column(name = "major")
	private String major;
    @Column(name = "gpa")
	private String gpa;
    @Column(name = "cover_letter")
	private String cover;
    @Column(name = "hobby")
	private String hobby;
    @Column(name = "gender")
	private String gender;
    @Column(name = "religion")
	private String religion;

    @OneToOne
    private DetailBiodata detailBiodata;


}
